core/src/main/java/org/elasticsearch/common/xcontent/ObjectSpooler.java

```diff
@@ -225,29 +217,35 @@ private void constructorArg(int position, ParseField parseField, Object value) {
             }
             constructorArgs[position] = value;
             constructorArgsCollected++;
-            if (constructorArgsCollected == constructorArgNames.size()) {
-                try {
-                    unspool();
-                } catch (ParsingException e) {
-                    throw new ParsingException(e.getLineNumber(), e.getColumnNumber(),
-                            "failed to build [" + objectParser.getName() + "] after last required field arrived", e);
-                } catch (Exception e) {
-                    throw new ParsingException(null, "Failed to build [" + objectParser.getName() + "] after last required field arrived",
-                            e);
+            if (constructorArgsCollected != constructorArgNames.size()) {
+                return;
+            }
+            try {
+                targetObject = builder.apply(constructorArgs);
+                while (spooledFields > 0) {
+                    spooledFields -= 1;
+                    spooled[spooledFields].accept(targetObject);
                 }
+            } catch (ParsingException e) {
+                throw new ParsingException(e.getLineNumber(), e.getColumnNumber(),
+                        "failed to build [" + objectParser.getName() + "] after last required field arrived", e);
+            } catch (Exception e) {
+                throw new ParsingException(null, "Failed to build [" + objectParser.getName() + "] after last required field arrived", e);
             }
         }
 
         /**
          * Spool up a consumer that we'll call once the targetObject is built. If targetObject has been built just apply the consumer
          * immediately.
          */
+        @SuppressWarnings("unchecked")
```

Comment:
Can we do something like

```diff
diff --git a/core/src/main/java/org/elasticsearch/common/xcontent/ConstructingObjectParser.java b/core/src/main/java/org/elasticsearch/common/xcontent/ConstructingObjectParser.java
index b8fe09a..8d6b3ca 100644
--- a/core/src/main/java/org/elasticsearch/common/xcontent/ConstructingObjectParser.java
+++ b/core/src/main/java/org/elasticsearch/common/xcontent/ConstructingObjectParser.java
@@ -244,11 +244,12 @@ public final class ConstructingObjectParser<Value, Context> extends AbstractObje
          * Queue a consumer that we'll call once the targetObject is built. If targetObject has been built this will fail because the caller
          * should have just applied the consumer immediately.
          */
-        @SuppressWarnings("unchecked")
+
         private void queue(Consumer<Value> queueMe) {
             assert targetObject == null: "Don't queue after the targetObject has been built! Just apply the consumer directly.";
             if (queuedFields == null) {
-                this.queuedFields = new Consumer[numberOfFields];
+                @SuppressWarnings("unchecked") Consumer<Value>[] queuedFields = new Consumer[numberOfFields];
+                this.queuedFields = queuedFields;
             }
             queuedFields[queuedFieldsCount] = queueMe;
             queuedFieldsCount++;
```

so that the suppression applies to a single line instead of the whole method?